server/apiv2/handlers/members.go

```diff
@@ -0,0 +1,215 @@
+			}
+		}
+		c.AbortWithStatus(http.StatusNotFound)
+	}
+}
+
+type updateMembersParams struct {
+	LeaderPriority float64 `json:"leader_priority"`
+}
+
+// UpdateMemberByName will update the PD member info according to the given parameters.
+func UpdateMemberByName() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		svr := c.MustGet("server").(*server.Server)
+
+		// Get etcd ID by name.
+		id, err := getMemberIDByName(svr, c.Param("name"))
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+		if id == 0 {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+
+		var p updateMembersParams
+		if err := c.BindJSON(&p); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrBindJSON.Wrap(err).GenWithStackByCause())
+			return
+		}
+		err = svr.GetMember().SetMemberLeaderPriority(id, int(p.LeaderPriority))
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+		c.JSON(http.StatusOK, nil)
+	}
+}
+
+// DeleteMemberByName will delete the PD member according to the given member's name.
+func DeleteMemberByName() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		svr := c.MustGet("server").(*server.Server)
+
+		// Get etcd ID by name.
+		id, err := getMemberIDByName(svr, c.Param("name"))
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		if id == 0 {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+
+		if err := deleteMemberByID(svr, id); err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+		c.JSON(
+			http.StatusOK,
+			nil,
+		)
+	}
+}
+
+func getMemberIDByName(svr *server.Server, name string) (id uint64, err error) {
+	members, err := svr.GetMembers()
+	if err != nil {
+		return
+	}
+	for _, m := range members {
+		if name == m.Name {
+			id = m.GetMemberId()
+			break
+		}
+	}
+	return
+}
+
+func deleteMemberByID(svr *server.Server, id uint64) error {
+	// Delete config.
+	err := svr.GetMember().DeleteMemberLeaderPriority(id)
+	if err != nil {
+		return err
+	}
+
+	// Delete dc-location info.
+	err = svr.GetMember().DeleteMemberDCLocationInfo(id)
+	if err != nil {
+		return err
+	}
+
+	client := svr.GetClient()
+	_, err = etcdutil.RemoveEtcdMember(client, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getAllMembers(svr *server.Server) (*pdpb.GetMembersResponse, error) {
+	req := &pdpb.GetMembersRequest{Header: &pdpb.RequestHeader{ClusterId: svr.ClusterID()}}
+	grpcServer := &server.GrpcServer{Server: svr}
+	members, err := grpcServer.GetMembers(context.Background(), req)
+	if err != nil {
+		return nil, err
+	}
+	dclocationDistribution, err := svr.GetTSOAllocatorManager().GetClusterDCLocationsFromEtcd()
+	if err != nil {
+		return nil, err
+	}
+	for _, m := range members.GetMembers() {
+		m.DcLocation = ""
+		binaryVersion, e := svr.GetMember().GetMemberBinaryVersion(m.GetMemberId())
```

Comment:
how about use `m.BinaryVersion` directly? So we needn't worry about missing L178?
```
          
  var e error
  m.BinaryVersion, e = svr.GetMember().GetMemberBinaryVersion(m.GetMemberId())
```